src/hooks/swap/useSyncTokenDefaults.ts

```diff
@@ -40,45 +42,73 @@ function useDefaultToken(
   return token ?? undefined
 }
 
-export default function useSyncTokenDefaults({
-  defaultInputTokenAddress,
-  defaultInputAmount,
-  defaultOutputTokenAddress,
-  defaultOutputAmount,
-}: TokenDefaults) {
-  const updateSwap = useUpdateAtom(swapAtom)
-  const { chainId } = useWeb3React()
-  const onSupportedNetwork = useOnSupportedNetwork()
+function useDefaultInputToken(
+  defaultAddress: DefaultAddress | undefined,
+  chainId: number | undefined,
+  defaultOutputToken: Currency | undefined
+): Currency | undefined {
   const nativeCurrency = useNativeCurrency()
-  const defaultOutputToken = useDefaultToken(defaultOutputTokenAddress, chainId)
-  const defaultInputToken =
-    useDefaultToken(defaultInputTokenAddress, chainId) ??
+  const onSupportedNetwork = useOnSupportedNetwork()
+
+  return (
+    useDefaultToken(defaultAddress, chainId) ??
     // Default the input token to the native currency if it is not the output token.
     (defaultOutputToken !== nativeCurrency && onSupportedNetwork ? nativeCurrency : undefined)
+  )
+}
 
-  const setToDefaults = useCallback(() => {
-    const defaultSwapState: Swap = {
-      amount: '',
-      [Field.INPUT]: defaultInputToken,
-      [Field.OUTPUT]: defaultOutputToken,
-      type: TradeType.EXACT_INPUT,
-    }
-    if (defaultInputToken && defaultInputAmount) {
-      defaultSwapState.amount = defaultInputAmount.toString()
-    } else if (defaultOutputToken && defaultOutputAmount) {
-      defaultSwapState.type = TradeType.EXACT_OUTPUT
-      defaultSwapState.amount = defaultOutputAmount.toString()
-    }
-    updateSwap((swap) => ({ ...swap, ...defaultSwapState }))
-  }, [defaultInputAmount, defaultInputToken, defaultOutputAmount, defaultOutputToken, updateSwap])
-
+export default function useSyncTokenDefaults(
+  { defaultInputTokenAddress, defaultInputAmount, defaultOutputTokenAddress, defaultOutputAmount }: TokenDefaults,
+  defaultChainId?: SupportedChainId
+) {
   const lastChainId = useRef<number | undefined>(undefined)
+  const lastConnector = useRef<Connector | undefined>(undefined)
+  const updateSwap = useUpdateAtom(swapAtom)
+  const { chainId, connector } = useWeb3React()
+
+  const defaultOutputToken = useDefaultToken(defaultOutputTokenAddress, chainId)
```

Comment:
I don't think that `chainId` is used to select the appropriate token. See L34.

Reply:
Good call out! This was actually there from before, it's not an addition. I believed it used [here](https://github.com/Uniswap/widgets/blob/9ebc5881b3968c6ca2307f1eb3c6c952293569a6/src/hooks/swap/useSyncTokenDefaults.ts#L37) to check if we can grab the Native token of the chain.